backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropCallConvertors.kt

```diff
@@ -56,6 +57,14 @@ private class InteropCallContext(
 
     fun IrType.isStoredInMemoryDirectly(): Boolean =
             isPrimitiveType() || isUnsigned() || isVector()
+
+    fun IrType.isSupportedReference(): Boolean = isObjCObjectType()
+            || isNullableString()
+            || classOrNull == symbols.list
+            || classOrNull == symbols.mutableList
+            || classOrNull == symbols.set
+            || classOrNull == symbols.map
+            || isFunction()
```

Comment:
The entire logic of determining and handling types in this file becomes similar to what `CBridgeGen` does. Is it possible to reuse some of the code?

Reply:
6d13354955feb7f34a091ba8f9ef36b296aef685